src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php

```diff
@@ -77,9 +77,18 @@ public function __construct(ContainerInterface $container, MessageSelector $sele
      */
     public function warmUp($cacheDir)
     {
+        // skip warmUp when translator doesn't use cache
+        if (!$currentDir = $this->options['cache_dir']) {
+            return;
+        }
+
+        $currentDir = $this->options['cache_dir'];
+        $this->options['cache_dir'] = $cacheDir;
         foreach ($this->resourceLocales as $locale) {
             $this->loadCatalogue($locale);
```

Comment:
Not sure if this can throw an exception? 

In that case, you might want to have a `catch` block to reset $this->options as well.